src/components/Global/Filter/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import RadioButton from '../RadioButton';
+import styles from './Filter.module.scss';
+
+type FilterProps = {
+  setFilterFields: React.Dispatch<React.SetStateAction<object>>;
+};
+const Filter = ({ setFilterFields }: FilterProps) => {
+  const [easyChecked, setEasyChecked] = useState(false);
+  const [mediumChecked, setMediumChecked] = useState(false);
+  const [hardChecked, setHardChecked] = useState(false);
+
+  return (
+    <div>
+      <form className={styles.filter}>
+        <RadioButton
+          id={'radioEasy'}
+          name={'filter'}
+          difficulty={'easy'}
+          checked={easyChecked}
+          onClick={() => {
+            setEasyChecked(wasChecked => !wasChecked);
+            setFilterFields({
```

Comment:
This is not how we want to do this, ideally we need to set up a `useEffect` hook that listens for changes to `easyChecked`, `mediumChecked`, and `hardChecked` and then uses `setFilterFields()` to match their values. Look into `useEffect()` for implementation details